crates/recursor/src/recursor_dns_handle.rs

```diff
@@ -328,37 +323,73 @@ impl RecursorDnsHandle {
             }
         }
 
-        // collect missing IP addresses, select over them all, get the addresses
-        // make it configurable to query for all records?
+        // If we have no glue, collect missing IP addresses for non-child NS servers
```

Comment:
I'm a little hesitant to actually ask for this since we've already done so many iterations, but: I think this commit could benefit from being split up again, with first the change to introduce `append_ips_from_lookup()` (and converting all the existing callers to that), and only then introducing the actual fix. That would make it a lot easier to see the semantic changes being introduced here.

Only if you're up for it, I'll approve this as is.

Reply:
No problem, I split the recursion fix into a different commit.

Reply:
Awesome!